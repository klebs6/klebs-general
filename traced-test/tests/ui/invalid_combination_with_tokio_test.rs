use traced_test::traced_test;

#[traced_test]
#[tokio::test]
async fn invalid_combination_of_test_attributes_with_tokio_test() {
    // This should never compile if another test attribute is present.
    // Test will fail if `#[tokio::test]` is present alongside `#[traced_test]`.
}

