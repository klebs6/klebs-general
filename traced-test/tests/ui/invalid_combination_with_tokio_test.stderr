error: custom attribute panicked
 --> tests/ui/invalid_combination_with_tokio_test.rs:4:1
  |
4 | #[traced_test]
  | ^^^^^^^^^^^^^^
  |
  = help: message: traced_test generator error: TokenStream(
              TokenStream [
                  Ident {
                      ident: "compile_error",
                      span: #5 bytes(109..123),
                  },
                  Punct {
                      ch: '!',
                      spacing: Alone,
                      span: #5 bytes(109..123),
                  },
                  Group {
                      delimiter: Parenthesis,
                      stream: TokenStream [
                          Literal {
                              kind: Str,
                              symbol: "The `traced_test` attribute should be used in place of `#[test]` or `#[tokio::test]`, not alongside them.",
                              suffix: None,
                              span: #5 bytes(109..123),
                          },
                      ],
                      span: #5 bytes(109..123),
                  },
                  Punct {
                      ch: ';',
                      spacing: Alone,
                      span: #5 bytes(109..123),
                  },
              ],
          )
